CONTRIBUTING.md

```diff
@@ -9,47 +9,46 @@ We welcome contributions from the community. Here are some guidelines.
 
 # Coding style
 
-* The Envoy source code is formated using clang-format. Thus all white space, etc. 
-  issues are taken care of automatically. The Travis tests will automatically check
-  the code format and fail. There are make targets that can both check the format 
-  (check_format) as well as fix the code format for you (fix_format).
-* Beyond code formatting, for the most part Envoy uses the 
-  [Google C++ style guidelines](https://google.github.io/styleguide/cppguide.html). 
-  The following section covers the major areas where we deviate from the Google 
-  guidelines.
-
-# Deviations from Google C++ Style guidelines
-
-* Exceptions are allowed and encouraged where appropriate.
-* References are always preferred over pointers when the reference cannot be null. This
-  includes both const and non-const references.
-* Function names using camel case starting with a lower case letter (e.g., "doFoo()").
-* Struct/Class member variables have a '\_' postfix (e.g., "int foo\_;").
-* 100 columns is the line limit.
-* Global static non-pod variables are allowed because Envoy correctly joins all threads on exit.
-  However, care is needed during init and static accessor functions may be required.
-* OOM events (both memory and FDs) are considered fatal crashing errors.
-* Use your GitHub name in TODO comments, e.g. `TODO(foobar): blah`.
-* Smart pointers are type aliased:
-  * `typedef std::unique_ptr<Foo> FooPtr;`
-  * `typedef std::shared_ptr<Bar> BarSharedPtr;`
-  * `typedef std::shared_ptr<const Blah> BlahConstSharedPtr;`
-  * Regular pointers (e.g. `int* foo`) should not be type aliased.
-* API-level comments should follow normal Doxygen conventions. Use `@param` to describe
-  parameters, `@return <return-type>` for return values.
-* There are probably a few other things missing from this list. We will add them as they
-  are brought to our attention.
+* See [STYLE.md](STYLE.md)
+
+# Breaking configuration change policy
+
+* As of the 1.3.0 release, the Envoy configuration is locked and we will not make breaking changes
+  between releases.
+* We reserve the right to deprecate configuration, and at the beginning of the following release
+  cycle remove the deprecated configuration.
+* All deprecations/breaking changes will be clearly listed in the release notes.
```

Comment:
This is on configuration, but what about other aspects of Envoy's backward compatibility story. For example:

* Flags: https://github.com/lyft/envoy/issues/613, although @rlazarus has a story for backwards compat there.
* APIs: The API aspect (e.g. REST endpoints) as opposed to payload (config) in RDS/CDS/SDS etc.
* Behavioral: Do we have a philosophy where we don't break the semantics of the config? E.g. in one version we prioritize some header in the processing pipeline over another, in the next we do it the other way.